### ISSUE TITLE
Is it possible to make DRM work?

### ISSUE THREAD

**Issue by flipphos** [webleon/Hackintosh-OptiPlex-7070-SFF#17]
I have an Asrock deskmini Z370 with 9600K and RX480 4G (MXM), on which DRM works under 11.5.x, with the parameter shikivga=80 in the DeviceProperties on the RX480 card, and SMBIOS 19,1.
It seems to me that this OptiPlex-7070-SFF/MT has the similar hardware (mine is 9600K, RX550, and iMac 19,1), but I didn't succeed to pass the FairPlay 1.x test and Apple TV+ test, though there is no problem to play Netflix, HBO etc in Chrome.

Did you succeed on the DRM?

**Comment by aya4dw** [comment]
Me too,  cannot  pass the FairPlay 1.x test. I would try to follow the instructions in the [artical](https://dortania.github.io/OpenCore-Post-Install/universal/drm.html) to have a try.

**Comment by webleon** [comment]
I can play DRM videos in Chrome and Edge, but not in Safari.
By the way, most RX550s a not original supported by MacOS, you can try to inject proper shikivga to iGPU see if it works.
https://github.com/acidanthera/WhateverGreen/blob/master/Manual/FAQ.Chart.md

**Comment by flipphos** [comment]
My RX550 is Polaris 21 based and works OOB in Big Sur. I have several of them working just fine.
The old RX550 is Baffin based, and needs to fake ID to make it work.

**Comment by flipphos** [comment]
By the way, I have the same RX550 on a Gigabyte Z97N and the DRM works in Big Sur.

**Comment by webleon** [comment]
Have you tried to disable ACPI patches in the config?

**Comment by flipphos** [comment]
There are 3 ACPI patches, which one(s) would you think is(are) relevant to try?
HPET _CRS to XCRS Rename
RTC IRQ 8 Patch
TIMR IRQ 0 Patch

**Comment by webleon** [comment]
All of them are belong to SSDT-HPET which only for fixing onboard sound card conflict.

**Comment by aya4dw** [comment]
Fixing DRM support and iGPU performance
Note: Safari 14 and macOS 11, Big Sur are currently unsupported by WhateverGreen's DRM patches. Safari 13 in Catalina and older are supported just fine however.
Note 2: Browsers not using hardware based DRM (ie. Mozilla Firefox or Chromium-based browsers like Google Chrome and Microsoft Edge) will have working DRM without any work both on iGPUs and dGPUs. The below guide is for Safari and other applications using hardware-based DRM.

**Comment by aya4dw** [comment]
In terminal, I entered "defaults write com.apple.AppleGVA gvaForceAMDKE -boolean yes".
Now I can watch stream DRM movie in apple TV app.

But "AMD GPU video decoder preference can be chosen through preferences overrides for some types of DRM content (like Apple TV and iTunes movie streaming). This preference may not always be compatible with the rest of the operating system and may cause problems with other ways of hardware media decoding and encoding. For this reason such an override is not recommended for daily use and shall only be enabled on demand."

List of overrides:

defaults write com.apple.AppleGVA gvaForceAMDKE -boolean yes forces AMD DRM decoder for streaming services (like Apple TV and iTunes movie streaming)
defaults write com.apple.AppleGVA gvaForceAMDAVCDecode -boolean yes forces AMD AVC accelerated decoder
defaults write com.apple.AppleGVA gvaForceAMDAVCEncode -boolean yes forces AMD AVC accelerated encoder
defaults write com.apple.AppleGVA gvaForceAMDHEVCDecode -boolean yes forces AMD HEVC accelerated decoder
defaults write com.apple.AppleGVA disableGVAEncryption -string YES forces AMD HEVC accelerated decoder
defaults write com.apple.coremedia hardwareVideoDecoder -string force forces hardware accelerated video decoder (for any resolution)
defaults write com.apple.coremedia hardwareVideoDecoder -string disable disables hardware accelerated video decoder (in QuickTime / Apple TV)

**Comment by flipphos** [comment]
Yes, I can confirm the command works on both of my 7070s. Neither could stream Apple TV+ movies before, and after executing the command, both can now.

I have 7070 MT + RX 580, and 7070 SFF + RX 550 (Polaris 21)

Still, I can’t pass the  iTunes movie FairPlay 1.x test.
